conjureup/juju.py

```diff
@@ -190,50 +190,52 @@ def add_config(k, v):
 
 
 def has_jaas_auth():
-    oauth_token = Path('~/.local/share/juju/store-usso-token').expanduser()
-    go_cookies = Path('~/.go-cookies').expanduser()
-    if oauth_token.exists():
-        return True
-    if go_cookies.exists():
-        go_cookies = json.loads(go_cookies.read_text())
-        for cookie in go_cookies or []:
+    jaas_cookies = Path('~/.local/share/juju/cookies/jaas.json').expanduser()
+    if jaas_cookies.exists():
+        jaas_cookies = json.loads(jaas_cookies.read_text())
+        for cookie in jaas_cookies or []:
             if cookie['Domain'] == consts.JAAS_DOMAIN:
-                return True
+                return bool(cookie['Value'])
     return False
 
 
 async def register_controller(name, endpoint, email, password, twofa,
                               timeout=30, fail_cb=None, timeout_cb=None):
+    app.log.info('Registering controller {}'.format(name))
     proc = await asyncio.create_subprocess_exec(
         'juju', 'register', '-B', endpoint,
         stdin=PIPE, stdout=PIPE, stderr=PIPE,
     )
-    if has_jaas_auth():
-        # if the user already authed with jujucharms.com, such as by
-        # logging in with the charm command, or registering JaaS and
-        # then unregistering it, we only need to name the controller
-        input = [name]
-    else:
-        input = [email, password, twofa, name]
     try:
-        stdin = b''.join(b'%s\n' % bytes(f, 'utf8') for f in input)
+        stdin = b''.join(b'%s\n' % bytes(f, 'utf8')
+                         for f in [name, email, password, twofa])
         stdout, stderr = await asyncio.wait_for(proc.communicate(stdin),
                                                 timeout)
         stdout = stdout.decode('utf8')
         stderr = stderr.decode('utf8')
+
+        prefix = 'Enter a name for this controller: '
```

Comment:
We provide input for it, but Juju has started putting the prompt for it on stderr instead of stdout.  All of the other prompts go on stdout like they should.  We should probably file a bug, but it doesn't seem like it'd be high priority and the work-around isn't terrible.